pkg/operator/operator_config.go

```diff
@@ -280,6 +299,68 @@ func (r *operatorConfigReconciler) ensureRuleEvaluatorSecrets(ctx context.Contex
 	return nil
 }
 
+// ensureAlertmanagerConfigSecret copies the managed Alertmanager config secret from gmp-public
+func (r *operatorConfigReconciler) ensureAlertmanagerConfigSecret(ctx context.Context, spec *monitoringv1.ManagedAlertmanagerSpec) error {
+	logger, _ := logr.FromContext(ctx)
+
+	// This is the default, no-op secret config. If we find a user-defined config,
+	// we will overwrite the default data with the user's data.
+	// If we don't find a user config, we will still proceed with ensuring this
+	// default secret exists (so that the alertmanager pod doesn't crash due to no
+	// config found). This flow also handles user deletion/disabling of managed AM.
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        "alertmanager",
+			Namespace:   r.opts.OperatorNamespace,
+			Annotations: componentAnnotations(),
+			Labels:      alertmanagerLabels(),
+		},
+		Data: map[string][]byte{"config.yaml": []byte(alertManagerNoOpConfig)},
```

Comment:
Make `config.yaml` a file-level const and reference it here to simplify refactoring if we need to support arbitrary keys in the future.